### PR TITLE
Issue17: Fetch Manifest from URL

### PR DESCRIPTION
**Pull request by haydenkarl22**
Fixes issue #17 

My Acceptance Criteria for this issue is:

- Users can input a URL pointing to a JSON manifest.
- The system validates the URL
- The system displays an appropriate message for success or failure.

In this PR, I have created a text field in the tools section of tools.html where users can input a URL pointing to a JSON manifest, and the system validates and displays the URL with an appropriate message for success or failure. 

I still need to extensively test the acceptance of URLs, as lots of unique JSON URLs will need to be viewable and fetchable via this text field. Below is an attachment of what my contributions look like in a browser.

image

### CODE REVIEW

File: web/tools.html [hunk 1]
```diff
@@ -95,5 +103,44 @@ <h4>Tools</h4>
...
+                            if (!response.ok) {
+                                throw new Error('Network response was not ok');
+                            }
+                            return response.json();
+                        })
+                        .then(data => {
+                            // process JSON URL data
+                            manifestMessage.textContent = 'Manifest loaded successfully!';
+                            manifestMessage.style.color = 'green';
+                            console.log(data); // only logging the manifest for now. this can be changed to open a specific tool to use or however we want to handle the newly loaded manifest.
```
Comment (sesheppard) [inline web/tools.html:135]: Is this currently the only thing done with the manifest data, or is it also saved to a variable somewhere?

File: web/tools.html [hunk 2]
```diff
@@ -95,5 +103,44 @@ <h4>Tools</h4>
             })
             .catch(error => console.error('Error loading footer:', error));
         </script>
+
```
Comment (sesheppard) [inline web/tools.html:106]: Should this functionality be moved to one of the other existing js files, or even to its own file entirely?

File: web/styles.css [hunk 3]
```diff
@@ -115,6 +115,46 @@ h4 {
     margin-top: 20px;
 }
 
+.manifest-loader {
+    max-width: 600px;
+    margin: 20px auto;
+    text-align: left;
+}
+
+.input-group {
```
Comment (sesheppard) [inline web/styles.css:124]: Is there any way we could clean up our styles at some point by identifying any classes with identical attributes and consolidating them?

File: web/tools.html [hunk 4]
```diff
@@ -95,5 +103,44 @@ <h4>Tools</h4>
...
+                    fetch(url)
+                        .then(response => {
+                            if (!response.ok) {
+                                throw new Error('Network response was not ok');
+                            }
+                            return response.json();
+                        })
+                        .then(data => {
+                            // process JSON URL data
```
Comment (haneenalsewari) [inline web/tools.html:132]: I feel like you could perform JSON structure validation here to ensure that it's a valid manifest before proceeding

**Review by alar12 — APPROVED** [review]
The functionality is working well as per requirement great work!!.

**Pull request merged**